### PR TITLE
Zanzana: Evaluate permissions alongside with RBAC engine

### PR DESCRIPTION
**Pull request by alexanderzobnin**
**What is this feature?**

Perform evaluation with zanzana if feature flag is enabled.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/738

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/accesscontrol/accesscontrol.go [hunk 1]
```diff
@@ -27,6 +29,10 @@ type AccessControl interface {
 	RegisterScopeAttributeResolver(prefix string, resolver ScopeAttributeResolver)
 }
 
+type ZanzanaChecker interface {
+	Check(ctx context.Context, in *openfgav1.CheckRequest) (*openfgav1.CheckResponse, error)
+}
```
Comment (kalleep) [inline pkg/services/accesscontrol/accesscontrol.go:34]: This is unused right?

File: pkg/services/accesscontrol/evaluator.go [hunk 2]
```diff
@@ -80,6 +84,25 @@ func match(scope, target string) bool {
 	return scope == target
 }
 
+func (p permissionEvaluator) EvaluateCustom(fn CheckerFn) (bool, error) {
```
Comment (kalleep) [inline pkg/services/accesscontrol/evaluator.go:87]: We can probably replace `Evaluate` with this and run extract the match function but lets leave it for later

**Review by kalleep — APPROVED** [review]
LGTM, I think there is a interface we can remove

File: pkg/services/accesscontrol/accesscontrol.go (hunk 1 above)
Comment (alexanderzobnin) [reply pkg/services/accesscontrol/accesscontrol.go:34]: yes, I forgot to remove it

**Review by IevaVasiljeva — APPROVED** [review]
Nice, LGTM!

**Review by Jguer — APPROVED** [review]

File: pkg/services/accesscontrol/acimpl/accesscontrol.go [hunk 3]
```diff
@@ -92,6 +96,10 @@ func (a *AccessControl) evaluateZanzana(ctx context.Context, user identity.Reque
 	}
 
 	return eval.EvaluateCustom(func(action, scope string) (bool, error) {
+		if implemented := zanzana.CheckAvailableAction(action); !implemented {
+			return false, ErrAccessNotImplemented
+		}
```
Comment (kalleep) [inline pkg/services/accesscontrol/acimpl/accesscontrol.go:101]: Not sure if we need `CheckAvailableAction`, translation should fail a couple of lines down and we can just return the error there then

File: pkg/services/accesscontrol/acimpl/accesscontrol.go [hunk 4]
```diff
@@ -16,6 +16,10 @@ import (
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
 )
 
+var (
+	ErrAccessNotImplemented = errors.New("access control not implemented for resource")
```
Comment (kalleep) [inline pkg/services/accesscontrol/acimpl/accesscontrol.go:20]: No need to export

**Pull request merged**